providers/apis/coinbase/api_handler.go

```diff
@@ -8,103 +8,115 @@ import (
 type APIHandler struct {
-	// cfg is the config for the Coinbase API.
-	cfg config.ProviderConfig
+	// market is the config for the Coinbase API.
+	market mmtypes.MarketConfig
+	// api is the config for the Coinbase API.
+	api config.APIConfig
 }
 
-// NewAPIHandler returns a new Coinbase APIDataHandler.
+// NewAPIHandler returns a new Coinbase PriceAPIDataHandler.
 func NewAPIHandler(
-	cfg config.ProviderConfig,
-) (handlers.APIDataHandler[slinkytypes.CurrencyPair, *big.Int], error) {
-	if err := cfg.ValidateBasic(); err != nil {
-		return nil, fmt.Errorf("invalid provider config %w", err)
+	market mmtypes.MarketConfig,
+	api config.APIConfig,
+) (types.PriceAPIDataHandler, error) {
+	if err := market.ValidateBasic(); err != nil {
+		return nil, fmt.Errorf("invalid market config for %s: %w", Name, err)
 	}
 
-	if !cfg.API.Enabled {
-		return nil, fmt.Errorf("api is not enabled for provider %s", cfg.Name)
+	if market.Name != Name {
+		return nil, fmt.Errorf("expected market config name %s, got %s", Name, market.Name)
 	}
 
-	if cfg.Name != Name {
-		return nil, fmt.Errorf("expected provider config name %s, got %s", Name, cfg.Name)
+	if api.Name != Name {
+		return nil, fmt.Errorf("expected api config name %s, got %s", Name, api.Name)
+	}
+
+	if !api.Enabled {
+		return nil, fmt.Errorf("api config for %s is not enabled", Name)
+	}
+
+	if err := api.ValidateBasic(); err != nil {
+		return nil, fmt.Errorf("invalid api config for %s: %w", Name, err)
 	}
 
 	return &APIHandler{
-		cfg,
+		market: market,
+		api:    api,
 	}, nil
 }
 
 // CreateURL returns the URL that is used to fetch data from the Coinbase API for the
-// given currency pair. Since the Coinbase API only supports fetching spot prices for
-// a single currency pair at a time, this function will return an error if the currency
-// pair slice contains more than one currency pair.
+// given tickers. Since the Coinbase API only supports fetching spot prices for a single
+// ticker at a time, this function will return an error if the ticker slice contains more
+// than one ticker.
 func (h *APIHandler) CreateURL(
-	cps []slinkytypes.CurrencyPair,
+	tickers []mmtypes.Ticker,
 ) (string, error) {
-	if len(cps) != 1 {
-		return "", fmt.Errorf("expected 1 currency pair, got %d", len(cps))
+	if len(tickers) != 1 {
+		return "", fmt.Errorf("expected 1 ticker, got %d", len(tickers))
 	}
 
 	// Ensure that the base and quote currencies are supported by the Coinbase API and
 	// are configured for the handler.
-	cp := cps[0]
-	market, ok := h.cfg.Market.CurrencyPairToMarketConfigs[cp.String()]
+	ticker := tickers[0]
+	market, ok := h.market.TickerConfigs[ticker.String()]
 	if !ok {
-		return "", fmt.Errorf("unknown currency pair %s", cp)
+		return "", fmt.Errorf("unknown ticker %s", ticker.String())
 	}
 
-	return fmt.Sprintf(h.cfg.API.URL, market.Ticker), nil
+	return fmt.Sprintf(h.api.URL, market.OffChainTicker), nil
```

Comment:
Are we always ensuring that `h.api.URL` will be a valid format string?

i.e. a string containing "%s" or something in it?

Reply:
yea we do

Reply:
we are so smart